nav2_recoveries/include/nav2_recoveries/recovery.hpp

```diff
@@ -43,26 +44,20 @@ enum class Status : int8_t
 using namespace std::chrono_literals;  //NOLINT
 
 template<typename ActionT>
-class Recovery
+class Recovery : public nav2_core::Recovery
 {
 public:
-  using ActionServer = nav2_util::SimpleActionServer<ActionT>;
-
-  explicit Recovery(
-    rclcpp::Node::SharedPtr & node, const std::string & recovery_name,
-    std::shared_ptr<tf2_ros::Buffer> tf)
-  : node_(node),
-    recovery_name_(recovery_name),
-    tf_(*tf),
-    action_server_(nullptr),
-    cycle_frequency_(10)
+  using ActionServer = nav2_util::SimpleActionServer<ActionT, rclcpp_lifecycle::LifecycleNode>;
+
+  Recovery()
+  : action_server_(nullptr),
+    cycle_frequency_(10),
```

Comment:
Seems like this should be made a parameter.

Reply:
I agree, I actually have a ticket open to make this configurable per-recovery, but hasn't been a priority recently